modules/table/src/main/java/org/apache/ignite/internal/table/KeyValueViewImpl.java

```diff
@@ -133,7 +136,9 @@ public void putAll(@NotNull Map<K, V> pairs) {
     @Override
     public @NotNull
     CompletableFuture<Void> putAllAsync(@NotNull Map<K, V> pairs) {
-        throw new UnsupportedOperationException("Not implemented yet.");
+        Collection<BinaryRow> rows = marshal(Objects.requireNonNull(pairs));
+
+        return tbl.upsertAll(rows, tx);
```

Comment:
why 'upsertAll' instead of 'insertAll' ?

Reply:
Because put() updates an existed row as an upsert() does.
An insert() ignores existed row, like a putIfAbsent()

Reply:
yep!